nodejs/eks/authenticationMode.ts

```diff
@@ -34,24 +35,31 @@ export function validateAuthenticationMode(args: ClusterOptions) {
         );
     }
 
-    const configMapOnlyProperties: (keyof ClusterOptions)[] = [
-        "roleMappings",
-        "userMappings",
-        "instanceRoles",
-    ];
-    const apiOnlyProperties: (keyof ClusterOptions)[] = ["accessEntries"];
-
     if (!supportsConfigMap(args.authenticationMode)) {
-        configMapOnlyProperties.forEach((prop) => {
-            if (args[prop]) {
+        const check: (prop: string) => (propertyValue: any|undefined) => void = (prop) => (pv) => {
```

Comment:
Nit: `prop` could be `keyof ClusterOptions`. What do you think about renaming this to `checkIsUndefined`? That way it's clearer what it is actually checking